netplan_cli/cli/state.py

```diff
@@ -88,24 +88,20 @@ def __init__(self, ip: dict, nd_data: JSON = [], nm_data: JSON = [],
             for obj in _routes:
                 if obj.get('dev') == self.name:
                     elem = {'to': obj.get('dst')}
-                    val = obj.get('gateway')
-                    if val:
+                    if val := obj.get('gateway'):
```

Comment:
This refactoring seems to be unrelated to the PR, but it's fine landing it in combination with this PR.

This feature is supported since Python 3.8 (i.e. back to Ubuntu Focal).
https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions

suggestion (non-blocking): Isolating this refactoring into its own commit could be beneficial to separate the logical chunks.